src/bridges/tokemak/TokemakBridge.sol

```diff
@@ -0,0 +1,335 @@
+        loadTokens();
+    }
+
+    function convert(
+        AztecTypes.AztecAsset memory inputAssetA,
+        AztecTypes.AztecAsset memory inputAssetB,
+        AztecTypes.AztecAsset memory outputAssetA,
+        AztecTypes.AztecAsset memory outputAssetB,
+        uint256 totalInputValue,
+        uint256 interactionNonce,
+        uint64 auxData,
+        address rollupBeneficiary
+    )
+        external
+        payable
+        override
+        returns (
+            uint256 outputValueA,
+            uint256 outputValueB,
+            bool isAsync
+        )
+    {
+        // // ### INITIALIZATION AND SANITY CHECKS
+        if (msg.sender != rollupProcessor) revert InvalidCaller();
+        if (inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert InvalidAssetType();
+        if (outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert InvalidAssetType();
+
+        // Check whether the call is for withdrawal or deposit
+        bool isWithdrawal = auxData != 0;
+
+        if (isWithdrawal) {
+            if (assets[inputAssetA.erc20Address] != outputAssetA.erc20Address)
+                revert InvalidAsset(inputAssetA.erc20Address);
+        } else {
+            if (tTokens[inputAssetA.erc20Address] != outputAssetA.erc20Address)
+                revert InvalidAsset(inputAssetA.erc20Address);
+        }
+
+        address tAsset = isWithdrawal ? inputAssetA.erc20Address : tTokens[inputAssetA.erc20Address];
+
+        if (tAsset == address(0)) revert InvalidAsset(tAsset);
+
+        // Withdraw or Deposit
+        if (isWithdrawal) {
+            isAsync = true;
+            outputValueA = 0;
+            addWithdrawalNonce(interactionNonce, tAsset, totalInputValue);
+        } else {
+            isAsync = false;
+            outputValueA = deposit(tAsset, totalInputValue, inputAssetA.erc20Address);
+        }
+
+        finalisePendingInteractions(MIN_GAS_FOR_FUNCTION_COMPLETION);
+    }
+
+    function finalise(
+        AztecTypes.AztecAsset calldata inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata outputAssetA,
+        AztecTypes.AztecAsset calldata,
+        uint256 interactionNonce,
+        uint64 auxData
+    )
+        external
+        payable
+        override
+        returns (
+            uint256 outputValueA,
+            uint256 outputValueB,
+            bool interactionCompleted
+        )
+    {
+        if (msg.sender != rollupProcessor) revert InvalidCaller();
+        if (inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert InvalidAssetType();
+        if (outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20) revert InvalidAssetType();
+
+        // Pending withdrawal value
+        uint256 inputValue = pendingInteractions[interactionNonce].inputValue;
+        if (inputValue <= 0) revert InvalidInput();
+
+        address tAsset = inputAssetA.erc20Address;
+        if (tAsset == address(0)) revert InvalidAsset(tAsset);
```

Comment:
Please do this check above line 150. On line 150 you do 1 SLOAD which costs 5000 gas so it's a good practice to revert ASAP to save gas.